### PR TITLE
soc: ace_v1x: l2 mm driver corrections

### PR DESCRIPTION
**Pull request by jxstelter**
This PR introduces 2 corrections to the MTL TLB driver:
1. on driver init unmaps the entire unused virtual address space instead of remaining physical memory space
2. replaces the sys_mm_drv_move_region implementation to query TLB for physical address for each virtual page to be remapped instead of assuming linear mapping as it is done in sys_mm_drv_simple_move_region

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Review by pjdobrowolski — APPROVED** [review]

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c [hunk 1]
```diff
@@ -517,19 +517,99 @@ int sys_mm_drv_remap_region(void *virt_old, size_t size,
...
+
+out:
+	/*
+	 * Since move is done in virtual space, need to
 	 * flush the cache to make sure the backing physical
 	 * pages have the new data.
 	 */
-	z_xtensa_cache_flush(va_new, size);
+	z_xtensa_cache_flush(virt_new, size);
+	z_xtensa_cache_flush_inv(virt_old, size);
```
Comment (aborisovich) [inline drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:612]: Please don't use goto and labels its a recipe for spaghetti code.
You can just put k_spin_unlock(&sys_mm_drv_common_lock, key); to where goto unlock_out is and
Put
image
into some simple inline function call.

**Review by aborisovich — CHANGES_REQUESTED** [review]

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c (hunk 1 above)
Comment (aborisovich) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:612]: Dropping, my "innovative beliefs" are just my opinion. Using goto is not agains project code rules,

**Review by aborisovich — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c (hunk 1 above)
Comment (aborisovich) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:612]: image
Sorry, this is agains project coding rules.
see https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html .

**Review by aborisovich — CHANGES_REQUESTED** [review]

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c (hunk 1 above)
Comment (aborisovich) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:612]: Ok it's inside `sys_mm_drv_move_region` function body. DId not expand all changes - it looked at defined as global scope. My bad.

**Review by aborisovich — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**